gobblin-data-management/src/main/java/gobblin/data/management/copy/CopyableFile.java

```diff
@@ -0,0 +1,166 @@
+/*
+ * Copyright (C) 2014-2015 LinkedIn Corp. All rights reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use
+ * this file except in compliance with the License. You may obtain a copy of the
+ * License at  http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the License is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied.
+ */
+
+package gobblin.data.management.copy;
+
+import gobblin.configuration.ConfigurationKeys;
+import gobblin.data.management.partition.File;
+
+import java.io.ByteArrayInputStream;
+import java.io.ByteArrayOutputStream;
+import java.io.DataInput;
+import java.io.DataInputStream;
+import java.io.DataOutput;
+import java.io.DataOutputStream;
+import java.io.IOException;
+import java.util.List;
+import java.util.Properties;
+
+import lombok.AccessLevel;
+import lombok.AllArgsConstructor;
+import lombok.EqualsAndHashCode;
+import lombok.Getter;
+import lombok.NoArgsConstructor;
+import lombok.Setter;
+
+import org.apache.commons.codec.DecoderException;
+import org.apache.commons.codec.binary.Hex;
+import org.apache.hadoop.fs.FileStatus;
+import org.apache.hadoop.fs.Path;
+import org.apache.hadoop.io.Text;
+import org.apache.hadoop.io.Writable;
+
+import com.google.common.collect.Lists;
+
+
+/**
+ * Abstraction for a file to copy from {@link #origin} to {@link #destination}.
+ */
+@Getter
+@Setter
+@AllArgsConstructor
+@NoArgsConstructor(access = AccessLevel.PRIVATE)
+@EqualsAndHashCode
+public class CopyableFile implements File, Writable {
+
+  public static final String SERIALIZED_COPYABLE_FILE = "gobblin.copy.serialized.copyable.file";
+
+  /** {@link FileStatus} of the existing origin file. */
+  private FileStatus origin;
```

Comment:
I don't think this should be part of the object:

1) We don't want to force a FS operation to get the status for every single file (that can be expensive!)
2) File status may change over time. It is better to read it whenever needed. 

At the very best, the instance here should be just a cached copy which is populated on demand.